partitura/utils/generic.py

```diff
@@ -171,6 +171,35 @@ def iter_subclasses(cls, _seen=None):
                 yield sub
 
 
+def first_order_derivative(func, x0, dx=0.5):
+    """
+    Compute the first order derivative of a function at a point `x0` using
+    a central difference scheme. This function is an adaptation of the scipy _derivative
```

Comment:
I would add the link https://github.com/scipy/scipy/blob/v1.7.0/scipy/misc/common.py#L75-L145 
which should stay the same